### PR TITLE
Add tag property into Button to let user pick a DOM tag

### PR DESCRIPTION
**Pull request by oorestisime**
Closes: #2463
Signed-off-by: Orestis Ioannou <oorestisime@gmail.com>

#### What does this PR do?

Let user pick a tag for the Anchor component

#### What testing has been done on this PR?

I added a unit test changing the tag to `span`

#### What are the relevant issues?

#2463 

#### Do the grommet docs need to be updated?

I update the readme and doc file.

#### Is this change backwards compatible or is it a breaking change?

I kept current compatibility with Anchor when use does not supply a tag

### CODE REVIEW

**Comment by oorestisime** [comment]
wow lol sorry. i dont know why i added all this to button instead of Anchor. i am closing and reopening correct one later

**Comment by alansouzati** [comment]
No problem at all. I'm glad you find you way through the undocumented process of sending a contribution!

File: src/js/components/Button/Button.js [hunk 1]
```diff
@@ -70,13 +71,20 @@ class Button extends Component {
           ],
       });
     }
+
+    let domTag = tag;
```
Comment (grommetux) [inline src/js/components/Button/Button.js]: can we replace it with a ternary "if" ?

`const domTag = !tag && href ? 'a' : undefined;`

File: src/js/components/Button/__tests__/Button-test.js [hunk 2]
```diff
@@ -220,3 +220,13 @@ test('Button is clickable', () => {
   button[0].props.onClick();
   expect(onClick).toBeCalled();
 });
+
+test('renders tag', () => {
+  const component = renderer.create(
+    <Grommet>
+      <Button tag="span" />
```
Comment (alansouzati) [inline src/js/components/Button/__tests__/Button-test.js]: I'm sorry i did not mention this. but can we use `as` to align with styled-component and the other anchor pr ?

Comment (oorestisime) [reply src/js/components/Button/__tests__/Button-test.js]: Yes i am on it. Should i check other components as well? At least the Text component is using tag https://github.com/grommet/grommet/blob/master/src/js/components/Text/index.d.ts#L10

**Comment by alansouzati** [comment]
thanks for your contribution! welcome to the team 🎉

**Pull request merged**